lib/settings.js

```diff
@@ -0,0 +1,156 @@
+import _ from 'lodash';
+import logger from './logger';
+
+
+const SETTINGS_CAPS = [
+  'locationServicesEnabled',
+  'locationServicesAuthorized',
+];
+const SAFARI_SETTINGS_CAPS = [
+  'safariAllowPopups',
+  'safariIgnoreFraudWarning',
+  'safariOpenLinksInBackground',
+];
+
+async function launchAndQuitSimulator (sim, safari) {
+  logger.debug('No simulator directories found.');
+  return await sim.launchAndQuit(safari);
+}
+
+function checkPreferences (settings, opts = {}) {
+  for (let setting of settings) {
+    if (_.has(opts, setting)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+// pass in the simulator so that other systems that use the function can supply
+// whatever they have
+async function setLocale (sim) {
```

Comment:
I would love for most of these functions (indeed, most of the functions on the driver). I'm working on the reset logic, and will try to see if I can make them as functional as possible.